src/ast/semantic_analyser.cpp

```diff
@@ -1499,6 +1499,32 @@ void SemanticAnalyser::visit(Binop &binop)
           << " operator can not be used on expressions of types " << lhs << ", "
           << rhs;
     }
+    else if (binop.op == Parser::token::EQ && !binop.left->is_literal &&
```

Comment:
I feel like emitting a warning and always return false should be better here. Don't want to block a valid script.

Reply:
sure, will fix.